pkg/terminal/starbind/starlark.go

```diff
@@ -0,0 +1,286 @@
+package starbind
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"runtime"
+	"strings"
+	"sync"
+
+	"go.starlark.net/resolve"
+	"go.starlark.net/starlark"
+
+	"github.com/go-delve/delve/service"
+	"github.com/go-delve/delve/service/api"
+)
+
+//go:generate go run ../../../scripts/gen-starlark-bindings.go go ./starlark_mapping.go
+//go:generate go run ../../../scripts/gen-starlark-bindings.go doc ../../../Documentation/cli/starlark.md
+
+const (
+	dlvCommandBuiltinName        = "dlv_command"
+	readFileBuiltinName          = "read_file"
+	writeFileBuiltinName         = "write_file"
+	commandPrefix                = "command_"
+	dlvContextName               = "dlv_context"
+	curScopeBuiltinName          = "cur_scope"
+	defaultLoadConfigBuiltinName = "default_load_config"
+)
+
+func init() {
+	resolve.AllowNestedDef = true
+	resolve.AllowLambda = true
+	resolve.AllowFloat = true
+	resolve.AllowSet = true
+	resolve.AllowBitwise = true
+	resolve.AllowRecursion = true
+	resolve.AllowGlobalReassign = true
+}
+
+// Context is the context in which starlark scripts are evaluated.
+// It contains methods to call API functions, command line commands, etc.
+type Context interface {
+	Client() service.Client
+	RegisterCallback(name, helpMsg string, cmdfn func(args string) error)
```

Comment:
nit: This would be more clear as `RegisterCommand`.

Reply:
Done.